### ISSUE TITLE
Symbolic indexing of the solution of an OptimizationProblem is slow

### ISSUE THREAD

**Issue by ValentinKaisermayer** [SciML/ModelingToolkit.jl#1944]
The first time one accesses a solution via a symbol it is very slow, since `generated_observed` has to compile the generated function. This happens even if the system has no observed variables at all.

**Comment by ChrisRackauckas** [comment]
Yes, we should really construct a symbol map dictionary which then gets used. This isn't about OptimizationSystem but any AbstractSystem.

**Comment by ChrisRackauckas** [comment]
Fixed with the new SII
